superset-frontend/src/dashboard/components/FiltersBadge/selectors.ts

```diff
@@ -178,14 +174,15 @@ export const selectIndicatorsForChart = (
 };
 
 export const selectNativeIndicatorsForChart = (
-  nativeFilters: NativeFiltersState,
+  nativeFilters: Filters,
   dataMask: DataMaskStateWithId,
   chartId: number,
-  charts: any,
+  chart: any,
   dashboardLayout: Layout,
   chartConfiguration: ChartConfiguration = {},
 ): Indicator[] => {
-  const chart = charts[chartId];
+  // no indicators before chart is rendered
+  if (chart.chartStatus !== 'rendered') return [];
```

Comment:
This logic should probably be in a render function rather than in a selector

Reply:
👍